src/main/java/zd/zero/waifu/motivator/plugin/listeners/IdleEventListener.kt

```diff
@@ -0,0 +1,66 @@
+package zd.zero.waifu.motivator.plugin.listeners
+
+import com.intellij.ide.IdeEventQueue
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.ProjectManager
+import zd.zero.waifu.motivator.plugin.alert.AlertConfiguration
+import zd.zero.waifu.motivator.plugin.assets.VisualMotivationAssetProvider
+import zd.zero.waifu.motivator.plugin.assets.WaifuAssetCategory
+import zd.zero.waifu.motivator.plugin.motivation.VisualMotivationFactory
+import zd.zero.waifu.motivator.plugin.settings.ThemeSettingsListener
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorPluginState
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorState
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorState.Companion.DEFAULT_IDLE_TIMEOUT_IN_MINUTES
+import java.util.concurrent.TimeUnit
+
+class IdleEventListener : Runnable, Disposable {
+    private val messageBus = ApplicationManager.getApplication().messageBus.connect()
+
+    init {
+        val self = this
+        messageBus.subscribe(ThemeSettingsListener.THEME_SETTINGS_TOPIC, object : ThemeSettingsListener {
+            override fun themeSettingsUpdated(themeSettings: WaifuMotivatorState) {
+                IdeEventQueue.getInstance().removeIdleListener(self)
+                IdeEventQueue.getInstance().addIdleListener(self,
+                    TimeUnit.MILLISECONDS.convert(
+                        themeSettings.idleTimoutInMinutes,
+                        TimeUnit.MINUTES
+                    ).toInt()
+                )
+            }
+        })
+        IdeEventQueue.getInstance().addIdleListener(
+            this,
+            TimeUnit.MILLISECONDS.convert(
+                getCurrentTimoutInMinutes(),
+                TimeUnit.MINUTES
+            ).toInt()
+        )
```

Comment:
question: Why are we using the theme settings to set the timer?

Reply:
Lol, whups, I straight up copy pasted code from the doki theme. 😅 
Imma fix it. You'll probably see that more times in the future.

Reply:
Fixxed